### PR TITLE
Gui command redesign

### PR DESCRIPTION
**Pull request by porterjc**
Currently there are some missing javadocs and a lot of opportunities of code cleanup that Alia and Jack will be addressing over the next sprint, but otherwise things are good to go for progress. Observer pattern and command pattern have been implemented and dependencies have been largely decoupled(Note, the commands are referred to as actions and actionrecievers). Currently filterActionLister still exists and needs to be replaced with appropriate actions, but the deprecated methods used in that class may need to be addressed as well so its being left alone for the time being. GraphActionListener is a necessary evil and will need to be addressed in the near future but is currently in a better state than it previously was. ConfusionMatrixPanel is currently gutted and commented out intentionally as Jack plans to reimplement the class such that it will be able to support the ability to add n matrices for n data sets, and simply just to make the code less ugly. 

TLDR: It's still messy but you can make progress now. We will fix the rest soon.

### CODE REVIEW

File: src/main/java/com/metsci/laproc/uicomponents/GraphDisplayerMouseListener.java [hunk 1]
```diff
@@ -95,20 +88,21 @@ public void mouseReleased(GlimpseMouseEvent glimpseMouseEvent) {
...
-                values.get(MetricDescriptionConstants.truePositives),
-                values.get(MetricDescriptionConstants.falsePositives)},
-                new double[]{values.get(MetricDescriptionConstants.trueNegatives),
-                        values.get(MetricDescriptionConstants.falseNegatives)});
-
-        window.getPointInfoPanel().update(point);
-        window.repaint();
-        System.out.println("X: " + point.getX() + " Y: " + point.getY());
-        ret = point.getX();
+//        window.getConfusionMatrixPanel().updateConfusionMatrix(new double[]{
```
Comment (christophermalinosky) [inline src/main/java/com/metsci/laproc/uicomponents/GraphDisplayerMouseListener.java]: Comments

File: src/main/java/com/metsci/laproc/tools/DataSheetPanel.java [hunk 2]
```diff
@@ -48,11 +56,17 @@ public void setDataSheet(ClassifierDataSet data) {
...
 			defaults.put("Table.alternateRowColor", new Color(240, 240, 240));
 
-		this.add(scrollPane);
+		this.panel.add(scrollPane);
 		JButton newEvalSetButton = new JButton("Create New Eval Set");
-        NewEvalSetActionListener nesaInstance = new NewEvalSetActionListener(this.window, tableDisplayer);
-		newEvalSetButton.addActionListener(nesaInstance);
-		this.add(newEvalSetButton);
+
+		ActionListener listener = new ActionListener() {
```
Comment (christophermalinosky) [inline src/main/java/com/metsci/laproc/tools/DataSheetPanel.java]: This seems like an ActionListener that we might make enough to warrant its own class.

File: src/main/java/com/metsci/laproc/tools/DataSetPanel.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        UIDefaults defaults = UIManager.getLookAndFeelDefaults();
+        if (defaults.get("Table.alternateRowColor") == null)
+            defaults.put("Table.alternateRowColor", new Color(240, 240, 240));
+
+        JPanel buttonPanel = new JPanel();
+        buttonPanel.setLayout(new BoxLayout(buttonPanel, BoxLayout.X_AXIS));
+
+        JButton displaySetButton = new JButton("Display Set");
+
+        displaySetButton.addActionListener(new ActionListener() {
```
Comment (christophermalinosky) [inline src/main/java/com/metsci/laproc/tools/DataSetPanel.java]: Same as below.
